feat: update `node-eth-kzg` to 0.9.1

kevaundray opened the pull request:
**Motivation**

This includes the update for the spec changes added here: https://github.com/ethereum/consensus-specs/pull/4519

kevaundray committed:
update to 0.9.1

nflaig commented:
@kevaundray thanks for updating that, I was just looking into this too because in https://github.com/ChainSafe/lodestar/pull/8450 we added a hack to pass spec tests which seems incorrect and in https://github.com/ChainSafe/lodestar/pull/8440 we added another workaround to skip those tests

Let me remove them, we should pass spec tests on your branch without the workarounds now

nflaig committed:
Remove hacks to skip recover_cells_and_kzg_proofs tests

nflaig commented on packages/beacon-node/package.json:
```diff
@@ -129,7 +129,7 @@
     "@chainsafe/ssz": "^1.2.2",
     "@chainsafe/threads": "^1.11.3",
-    "@crate-crypto/node-eth-kzg": "0.8.0",
+    "@crate-crypto/node-eth-kzg": "0.9.1",
```
curious about 0.10.0 which includes https://github.com/crate-crypto/rust-eth-kzg/pull/433, don't we want it to be multithreaded?

kevaundray replied:
yep we do -- we debugged it futher on the users (an L2) machine following the release and it was because they were on a mainframe machine and wanted to control the exact amount of threads created. async/tokio will launch up a pool of threads which is what they were experiencing, I plan to revert that PR because they can now control the number of threads launched using an environment variable

kevaundray commented:
> @kevaundray thanks for updating that, I was just looking into this too because in #8450 we added a hack to pass spec tests which seems incorrect and in #8440 we added another workaround to skip those tests
> 
> Let me remove them, we should pass spec tests on your branch without the workarounds now

we went back and forth between whether the library should sort them or just assert that they are sorted -- just checked and we assert they are sorted, so I think we still need to sort them in lodestar -- the ignored test should now pass though

nflaig commented:
> we went back and forth between whether the library should sort them or just assert that they are sorted -- just checked and we assert they are sorted, so I think we still need to sort them in lodestar -- the ignored test should now pass though

yeah I reviewed the changes you did on kzg side, throwing an error works for us, we sort them before calling the library
https://github.com/ChainSafe/lodestar/blob/d8afb6dc391bc663f4047c480a7f4f39d166940f/packages/beacon-node/src/util/blobs.ts#L87-L88

nflaig reviewed (approved):
LGTM - spec tests are passing now without workarounds

wemeetagain commented:
:tada: This PR is included in [v1.36.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.36.0) :tada:
